feature_engine/selection/target_mean_encoding.py

```diff
@@ -0,0 +1,264 @@
+from sklearn.base import TransformerMixin, BaseEstimator
+from sklearn.utils.validation import check_is_fitted
+from sklearn.model_selection import train_test_split
+from sklearn.metrics import roc_auc_score
+from sklearn.metrics import r2_score
+from feature_engine.variable_manipulation import _find_all_variables
+from feature_engine.variable_manipulation import _find_categorical_variables
+from feature_engine.variable_manipulation import _find_numerical_variables
+from feature_engine.variable_manipulation import _define_variables
+from feature_engine.dataframe_checks import _is_dataframe
+from feature_engine import categorical_encoders as ce
+from feature_engine import discretisers as dsc
+
+import pandas as pd 
+
+class TargetMeanEncoderFeatureSelector(BaseEstimator, TransformerMixin):
+    """
+    TargetMeanEncoderFeatureSelector
+    ------------------
+        Description
+        -----------
+        Calculates the feature importance.
+
+        For each categorical variable:
+            1) Separate into train and test
+            2) Determine the mean value of the target within each label of the categorical variable using the train set
+            3) Use that mean target value per label as the prediction (using the test set) and calculate the roc-auc.
+
+        For each numerical variable:
+            1) Separate into train and test
+            2) Divide the variable into 100 quantiles
+            3) Calculate the mean target within each quantile using the training set 
+            4) Use that mean target value / bin as the prediction (using the test set) and calculate the roc-auc
+        
+        Implementation
+        --------------
+
+            Public methods
+            --------------
+                `fit(self, X, y)`
+                `transform(self)`
+                `fit_transform(self, X, y)`
+        
+    Parameters
+    ----------
+
+    variables: list, default=None
+        The list of variables to evaluate. If None, the transformer will evaluate all 
+        variables in the dataset associated with the variables_type.
+
+    scoring: string, default='roc_auc_score'
+        This indicates the metrics score to perform the feature selection.
+        The current support includes 'roc_auc_score' and 'r2_score'.
+
+
+    """
+
+    def __init__(self, variables=None, scoring='roc_auc_score'):
+        self.variables = _define_variables(variables)
+        self.scoring = scoring
+        self.X_test_enc = None
+        self.y_test = None
+
+    def fit(self, X, y, quantiles=5, test_size=0.3, random_state=0):
```

Comment:
the fit method must only take the parameters X and y, so it remains compatible with the Scikit-learn pipeline. We need to send all the other params to init.